### PR TITLE
refactor: lint errors; strings, undefinied nums, global vars

### PR DESCRIPTION
**Pull request by AlisonB319**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: apiserver/getdistrosigimageconfig.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ const (
 // GetDistroSigImageConfig endpoint for sig config for all distros in one shot.
 func (api *APIServer) GetDistroSigImageConfig(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	_, cancel := context.WithTimeout(ctx, 30*time.Second) //nolint:gomnd // variable for 30 is not needed.
```
Comment (alexeldeib) [inline apiserver/getdistrosigimageconfig.go:23]: this is odd, we should either use the context or eliminate all this code

File: apiserver/getlatestsigimageconfig.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ const (
 // GetLatestSigImageConfig endpoint for getting latest sig image reference.
 func (api *APIServer) GetLatestSigImageConfig(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	_, cancel := context.WithTimeout(ctx, 30*time.Second) //nolint:gomnd // variable for 30 is not needed.
```
Comment (alexeldeib) [inline apiserver/getlatestsigimageconfig.go:23]: same ?

File: pkg/agent/datamodel/sig_config.go [hunk 3]
```diff
@@ -58,106 +58,36 @@ func GetCloudTargetEnv(location string) string {
...
-	AKSCBLMarinerV2Gen2Kata,
-	AKSCBLMarinerV2Arm64Gen2,
-	AKSCBLMarinerV2Gen2TL,
-	AKSCBLMarinerV2KataGen2TL,
-}
-
 // IsContainerdSKU returns true if distro type is containerd-enabled.
 func (d Distro) IsContainerdDistro() bool {
-	for _, distro := range AvailableContainerdDistros {
+	availableContainerdDistros := []Distro{
```
Comment (alexeldeib) [inline pkg/agent/datamodel/sig_config.go:63]: can we keep these as variables but make them unexported instead of redefining per func?

if not this is okay

Comment (AlisonB319) [reply pkg/agent/datamodel/sig_config.go:63]: Yeah making it unexported still violates the no global variables rule

**Pull request closed without merging**